src/cli/configuration_builder_spec.ts

```diff
@@ -65,52 +66,114 @@ describe('Configuration', () => {
-      const featurePath = path.join(cwd, relativeFeaturePath)
-      await fsExtra.outputFile(featurePath, '')
-      const supportCodePath = path.join(cwd, 'features', 'a.js')
-      await fsExtra.outputFile(supportCodePath, '')
-      const argv = baseArgv.concat([relativeFeaturePath])
-
-      // Act
-      const {
-        featurePaths,
-        pickleFilterOptions,
-        supportCodePaths,
-      } = await ConfigurationBuilder.build({ argv, cwd })
+    describe('without esm', () => {
+      it('returns the appropriate feature and support code paths', async function () {
+        // Arrange
+        const cwd = await buildTestWorkingDirectory()
+        const relativeFeaturePath = path.join('features', 'nested', 'a.feature')
+        const featurePath = path.join(cwd, relativeFeaturePath)
+        await fsExtra.outputFile(featurePath, '')
+        const jsSupportCodePath = path.join(cwd, 'features', 'a.js')
+        await fsExtra.outputFile(jsSupportCodePath, '')
+        const mjsSupportCodePath = path.join(cwd, 'features', 'b.mjs')
+        await fsExtra.outputFile(mjsSupportCodePath, '')
+        const argv = baseArgv.concat([relativeFeaturePath])
+
+        // Act
+        const {
+          featurePaths,
+          pickleFilterOptions,
+          supportCodePaths,
+        } = await ConfigurationBuilder.build({ argv, cwd })
+
+        // Assert
+        expect(featurePaths).to.eql([featurePath])
+        expect(pickleFilterOptions.featurePaths).to.eql([relativeFeaturePath])
+        expect(supportCodePaths).to.eql([jsSupportCodePath])
+      })
+    })
 
-      // Assert
-      expect(featurePaths).to.eql([featurePath])
-      expect(pickleFilterOptions.featurePaths).to.eql([relativeFeaturePath])
-      expect(supportCodePaths).to.eql([supportCodePath])
+    describe('with esm', () => {
+      it('returns the appropriate feature and support code paths', async function () {
+        // Arrange
+        const cwd = await buildTestWorkingDirectory()
+        const relativeFeaturePath = path.join('features', 'nested', 'a.feature')
+        const featurePath = path.join(cwd, relativeFeaturePath)
+        await fsExtra.outputFile(featurePath, '')
+        const jsSupportCodePath = path.join(cwd, 'features', 'a.js')
+        await fsExtra.outputFile(jsSupportCodePath, '')
+        const mjsSupportCodePath = path.join(cwd, 'features', 'b.mjs')
+        await fsExtra.outputFile(mjsSupportCodePath, '')
+        const argv = baseArgv.concat([relativeFeaturePath, '--esm'])
+
+        // Act
+        const {
+          featurePaths,
+          pickleFilterOptions,
+          supportCodePaths,
+        } = await ConfigurationBuilder.build({ argv, cwd })
+
+        // Assert
+        expect(featurePaths).to.eql([featurePath])
+        expect(pickleFilterOptions.featurePaths).to.eql([relativeFeaturePath])
+        expect(supportCodePaths).to.eql([jsSupportCodePath, mjsSupportCodePath])
+      })
```

Comment:
I don't think we need nested feature tests for non-esm. Thoughts on the following:
```js
describe('path to a feature', () => {
  describe('without esm', () => {})
  describe('with esm and js support files', () => {})
  describe('with esm and mjs support files', () => {})
});

describe('path to a nested feature', () => {}) // unchanged
```

Reply:
Makes sense - updated per your suggestion.